### PR TITLE
Fix parseSortStatement

### LINKED ISSUE #1477: DB error sqlite_database.cc:338
Originally I thought it was connected to DB init issue, but it seems random( and it got better end of  the stick on me). 
Upon machines/service restart( in case of service it improved once, maybe) you either end up running all right or screwed.
If everything goes fine, both UI and server are up. If not, AV server is up, but web UI is gone

2021-05-26 22:09:31   info: Gerbera shutting down. Please wait...
2021-05-26 22:09:31   info: Gerbera exiting. Have a nice day.
2021-05-26 22:09:31   info: Gerbera UPnP Server 1.8.1 - http://gerbera.io/
2021-05-26 22:09:31   info: Copyright 2016-2021 Gerbera Contributors.
2021-05-26 22:09:31   info: Gerbera is free software, covered by the GNU General Public License version 2
2021-05-26 22:09:31   info: Loading configuration from: /home/gerbera/etc/config.xml
2021-05-26 22:09:31   info: Checking configuration...
2021-05-26 22:09:31   info: Configuration check succeeded.
2021-05-26 22:09:31   info: About to migrate metadata from mt_cds_object to mt_metadata
2021-05-26 22:09:31   info: No data will be removed from mt_cds_object
2021-05-26 22:09:31   info: Migrated metadata - object count: 0
2021-05-26 22:09:31   info: Loading 0 configuration items from database
2021-05-26 22:09:31   info: Loading autoscan location: /srv/video/movies; recursive: true, last_modified: 1970-01-01 01:00:01
2021-05-26 22:09:31   info: Loading autoscan location: /srv/video/staging; recursive: true, last_modified: 1970-01-01 01:00:01
2021-05-26 22:09:31   info: Loading autoscan location: /srv/video/tv_series; recursive: true, last_modified: 1970-01-01 01:00:01
2021-05-26 22:09:31   info: Initialising UPnP with interface: Y, port: 55555
2021-05-26 22:09:31   info: IPv4: Server bound to: XXXX:55555
2021-05-26 22:09:31   info: IPv6: Server bound to: XXXX:55555
2021-05-26 22:09:31   info: IPv6 ULA/GLA: Server bound to: :55556
2021-05-26 22:09:31   info: The Web UI can be reached by following this link: http://XXXX:55555/
2021-05-26 22:09:31  error: prematurely shutting down.
2021-05-26 22:09:31  error: No such object: [/builddir/build/BUILD/gerbera-1.8.1/src/database/sqlite3/sqlite_database.cc:338] select Error: [/builddir/build/BUILD/gerbera-1.8.1/src/database/sqlite3/sqlite_database.cc:515] waitForTask Error: SQLITE3: (1, : 1) near ",": syntax error
Query: SELECT "f"."id", "f"."ref_id", "f"."parent_id", "f"."object_type", "f"."upnp_class", "f"."dc_title", "f"."location", "f"."location_hash", "f"."metadata", "f"."auxdata", "f"."resources", "f"."update_id", "f"."mime_type", "f"."flags", "f"."part_number", "f"."track_number", "f"."service_id", "f"."bookmark_pos", "f"."last_modified", "rf"."upnp_class", "rf"."location", "rf"."metadata", "rf"."auxdata", "rf"."resources", "rf"."mime_type", "rf"."service_id", "as"."persistent" FROM "mt_cds_object" "f" LEFT JOIN "mt_cds_object" "rf" ON "f"."ref_id"="rf"."id" LEFT JOIN "mt_autoscan" "as" ON "as"."obj_id"="f"."id"  WHERE "f"."parent_id"=0 ORDER BY ("f"."object_type"=1) DESC, , "f"."dc_title"dc:title LIMIT 100000 OFFSET 0
error: near ",": syntax error

and WEB UI does not work.

Another restart:

2021-05-26 22:09:31  error: prematurely shutting down.
2021-05-26 22:09:31  error: No such object: [/builddir/build/BUILD/gerbera-1.8.1/src/database/sqlite3/sqlite_database.cc:338] select Error: [/builddir/build/BUILD/gerbera-1.8.1/src/database/sqlite3/sqlite_database.cc:515] waitForTask Error: SQLITE3: (1, : 1) near ",": syntax error
Query: SELECT "f"."id", "f"."ref_id", "f"."parent_id", "f"."object_type", "f"."upnp_class", "f"."dc_title", "f"."location", "f"."location_hash", "f"."metadata", "f"."auxdata", "f"."resources", "f"."update_id", "f"."mime_type", "f"."flags", "f"."part_number", "f"."track_number", "f"."service_id", "f"."bookmark_pos", "f"."last_modified", "rf"."upnp_class", "rf"."location", "rf"."metadata", "rf"."auxdata", "rf"."resources", "rf"."mime_type", "rf"."service_id", "as"."persistent" FROM "mt_cds_object" "f" LEFT JOIN "mt_cds_object" "rf" ON "f"."ref_id"="rf"."id" LEFT JOIN "mt_autoscan" "as" ON "as"."obj_id"="f"."id"  WHERE "f"."parent_id"=0 ORDER BY ("f"."object_type"=1) DESC, , "f"."dc_title"dc:title LIMIT 100000 OFFSET 0
error: near ",": syntax error
2021-05-26 22:09:31  error: prematurely shutting down.
2021-05-26 22:09:37   info: Gerbera UPnP Server 1.8.1 - http://gerbera.io/
2021-05-26 22:09:37   info: Copyright 2016-2021 Gerbera Contributors.
2021-05-26 22:09:37   info: Gerbera is free software, covered by the GNU General Public License version 2
2021-05-26 22:09:37   info: Loading configuration from: /home/gerbera/etc/config.xml
2021-05-26 22:09:37   info: Checking configuration...
2021-05-26 22:09:37   info: Configuration check succeeded.
2021-05-26 22:09:37   info: About to migrate metadata from mt_cds_object to mt_metadata
2021-05-26 22:09:37   info: No data will be removed from mt_cds_object
2021-05-26 22:09:37   info: Migrated metadata - object count: 0
2021-05-26 22:09:37   info: Loading 0 configuration items from database
2021-05-26 22:09:37   info: Loading autoscan location: /srv/video/movies; recursive: true, last_modified: 1970-01-01 01:00:01
2021-05-26 22:09:37   info: Loading autoscan location: /srv/video/staging; recursive: true, last_modified: 1970-01-01 01:00:01
2021-05-26 22:09:37   info: Loading autoscan location: /srv/video/tv_series; recursive: true, last_modified: 1970-01-01 01:00:01
2021-05-26 22:09:37   info: Initialising UPnP with interface: X, port: 55555
2021-05-26 22:09:37   info: IPv4: Server bound to: XXX:55556
2021-05-26 22:09:37   info: IPv6: Server bound to: XXXX:55555
2021-05-26 22:09:37   info: IPv6 ULA/GLA: Server bound to: :55556
2021-05-26 22:09:37   info: The Web UI can be reached by following this link: http://XXXX:55556/

use-transactions="no"

### PR DESCRIPTION
**Pull request by KarlStraussberger**
fixes #1477

### CODE REVIEW

File: src/database/sql_database.cc [hunk 1]
```diff
@@ -607,35 +607,33 @@ std::unique_ptr<std::vector<int>> SQLDatabase::getServiceObjectIDs(char serviceP
...
+    std::vector<std::string> sort;
+    for (auto&& seg : splitString(sortCrit, ',')) {
+        seg = trimString(seg);
+        bool desc = (seg[0] == '-') ? true : false;
+        if (seg[0] == '-' || seg[0] == '+') {
+            seg = seg.substr(1);
+        } else {
+            log_warning("Unknown sort direction '{}' in '{}'", seg, sortCrit);
+        }
+        bool defined = false;
```
Comment (neheb) [inline src/database/sql_database.cc:622]: loop seems inefficient. that if statement can be taken out I think.

Comment (KarlStraussberger) [reply src/database/sql_database.cc:622]: The sortCriteria looks like "+dc::title,-upnp::track_number". the first if checks for the signs (+/-) the second for the key (title etc.). Which could be moved outside without losing the context.

Comment (neheb) [reply src/database/sql_database.cc:622]: Something like:

```
auto it = std::find_if(xxx)
if (it != x.end())
```

Comment (KarlStraussberger) [reply src/database/sql_database.cc:622]: Valid point. Sometimes I'm going too straight. Thanks.

**Pull request merged**